packages/admin/cms-admin/src/dam/DataGrid/FolderDataGrid.tsx

```diff
@@ -148,6 +159,14 @@ const FolderDataGrid = ({
         },
     });
 
+    const [moveDamItemDialogState, setMoveDamItemDialogState] = React.useState<"selection" | "single" | null>(null);
+    const [moveSingleDamItem, setMoveSingleDamItem] = React.useState<{ id: string; type: "file" | "folder" } | null>(null);
```

Comment:
For me, it was rather hard to understand what this two states mean.

How about creating a single `damItemsToMove` state and use it for both the selection and a single move?

```ts
const [damItemsToMove, setDamItemsToMove] = React.useState<Array<{ id: string; type: "file" | "folder" }>>([]);
```

You could set the state in `onMoveSingleDamItem` and `onOpenMoveDialog`, and handle the array in the same way as a single value in `onChooseFolder`